### ISSUE TITLE
onDragStart events dont fire when dragging slider

### ISSUE THREAD

**Issue by pragmatic84** [express-labs/pure-react-carousel#155]
Forgive me if my description is a bit rubbish, I'm relatively new to React and I've been working with this Carousel for about a day now.

The problem I'm having is that I'd like to have some CSS transition effects occur on the 'active' slide.

I'm currently defining that active slide using an index in state and updating that on click, so far so good.

However when I drag the slider (which works fine), obviously no events are firing because I havent defined them. I did a little bit of digginf and found the onDragStart event from the React docs and added that to the Slide component within my Carousel.

However when dragging the slides around, the event isn't triggered. If I was to guess why it's because the Slide itself isn't actually being dragged?

My thinking is set dragEnabled to false, and trigger the next/previous button clicks using a ref?
Is this a sensible approach or is there a baked in solution I haven't seen yet?

**Comment by mrbinky3000** [comment]
Pure React Carousel already tracks the active slide for you.  Check the `<Slide>` component documentation.  No need for you to duplicate that in state.  Could cause problems having more than one source of truth.

You would want to attach your drag events (onDragStart, onDragMove, onDragEnd) to the `<Slider>` component's `trayProps` property.  Check out the documentation for trayProps here... https://www.npmjs.com/package/pure-react-carousel#slider-

**Comment by pragmatic84** [comment]
> Pure React Carousel already tracks the active slide for you. Check the `<Slide>` component documentation. No need for you to duplicate that in state. Could cause problems having more than one source of truth.
> 
> You would want to attach your drag events (onDragStart, onDragMove, onDragEnd) to the `<Slider>` component's `trayProps` property. Check out the documentation for trayProps here... https://www.npmjs.com/package/pure-react-carousel#slider-

Not sure I exactly follow, but first let me say thanks for getting back to me so quickly.

Let me copy the code I have and maybe you can see where I'm going wrong? Specifically around the render function where I've created a 'sliderProps' object.

```
import React, { Component } from 'react';
import { CarouselProvider, Slider } from 'pure-react-carousel';
import Slide from './Slide';
import {
  CarouselWrapper,
  SliderWrapper,
  StyledButtonBack,
  StyledButtonNext,
  StyledChevronLeft,
  StyledChevronRight,
} from './styled-components';
import { DummyCarouselData } from '../dummyData';
import 'pure-react-carousel/dist/react-carousel.es.css';

class Carousel extends Component {
  constructor(props) {
    super(props);

    this.state = {
      activeSlide: 2,
    };

    this.incrementActiveSlide = this.incrementActiveSlide.bind(this);
    this.decrementActiveSlide = this.decrementActiveSlide.bind(this);
  }

  componentDidMount() {
    // fetch actual slide data when we have it
  }

  onDragStart(e) {
    console.log('firing');
  }

  incrementActiveSlide() {
    const { activeSlide } = this.state;
    this.setState({
      activeSlide: activeSlide + 1,
    });
  }

  decrementActiveSlide() {
    const { activeSlide } = this.state;
    this.setState({
      activeSlide: activeSlide - 1,
    });
  }

  renderSlides() {
    const { activeSlide } = this.state;
    return DummyCarouselData.map(({ src, heading, tagline }, i) => (
      <Slide
        className={activeSlide === i ? 'active' : ''}
        src={src}
        heading={heading}
        tagline={tagline}
        index={i}
        key={i}
      />
    ));
  }

  render() {
    const sliderProps = {
      onDragStart: () => this.onDragStart,
    };
    return (
      <CarouselWrapper>
        <CarouselProvider
          naturalSlideWidth={150}
          naturalSlideHeight={200}
          totalSlides={DummyCarouselData.length}
          visibleSlides={5}
        >
          <SliderWrapper>
            <Slider trayProps={sliderProps}>{this.renderSlides()}</Slider>
            <StyledButtonBack onClick={this.decrementActiveSlide}>
              <StyledChevronLeft />
            </StyledButtonBack>
            <StyledButtonNext onClick={this.incrementActiveSlide}>
              <StyledChevronRight />
            </StyledButtonNext>
          </SliderWrapper>
        </CarouselProvider>
      </CarouselWrapper>
    );
  }
}

export default Carousel;

```

**Comment by mrbinky3000** [comment]
Sorry for the late reply.  Working on a solution.

**Comment by mrbinky3000** [comment]
I've created a new example in our example app, Example 11, that attaches nearly every type of React synthetic event to the trayProps property.  They all fire except for the drag events. 

I've learned that the drag events are not firing because the slider tag is not a "draggable" element.  Even though you can use your mouse or finger to drag slides left and right, the slider is not using the Browser's Drag and Drop API. 

https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API

Instead, it listens to mousedown, mousemove and mouseup (touchstart, touchmove, touchend, and touchcancel on touch devices) to simulate scrolling, not dragging.

I've renamed some class methods so that developers looking at our code don't think our "drag" methods have anything to do with React's drag methods.

Long story short, if you want to fire some sort of transition on "drag" you're going to need to add your own event handlers to the sliderTray.  I'd start by listening to a mouseDown followed by mouseMove to kick off your transition.

See #161 for the code in Example 11.
